common/util.go

```diff
@@ -537,6 +538,211 @@ func GetSizeOfMapStringToByteArray(input map[string][]byte) int {
 	return res + golandMapReserverNumberOfBytes
 }
 
+// GetSizeOfMapStringToByteArray returns approximate size of the history event taking into account byte arrays only now
+func GetSizeOfHistoryEvent(event *shared.HistoryEvent) uint32 {
+	if event == nil {
+		return 0
+	}
+
+	res := 0
+	switch *event.EventType {
+	case shared.EventTypeWorkflowExecutionStarted:
+		if event.WorkflowExecutionStartedEventAttributes == nil {
```

Comment:
we can assuming event type and attribute are guaranteed matched, so no need for such check. 
If this rule is violated, lots of place will NPE, just let it expose itself so team can quick fix.